src/al-sap/al_service_data_unit.cpp

```diff
@@ -62,25 +70,27 @@ std::vector<unsigned char> AlServiceDataUnit::serialize() const {
 
 // Deserialization method
 void AlServiceDataUnit::deserialize(const std::vector<unsigned char>& data) {
-    // Check minimum size (6 bytes each for source and destination MAC, plus 3 bytes for flags and fragment ID)
-    if (data.size() < 15) {
+    // Check minimum size (4 bytes for size, 6 bytes each for source and destination MAC, plus 3 bytes for flags and fragment ID)
+    if (data.size() < 19) {
         throw AlServiceException("Insufficient data to deserialize AlServiceDataUnit", PrimitiveError::DeserializationError);
     }
```

Comment:
Similar here. I'm assuming the 4 bytes are an 32 bit unsigned int but it's not clear so along with the comment you can do:
```suggestion
    // Check minimum size (4 bytes for size, 6 bytes each for source and destination MAC, plus 3 bytes for flags and fragment ID)
    size_t min_size = sizeof(uint32_t) + 6 + 6 + sizeof(uint8_t) + sizeof(uint8_t) + sizeof(uint8_t);
    if (data.size() < min_size) {
        throw AlServiceException("Insufficient data to deserialize AlServiceDataUnit", PrimitiveError::DeserializationError);
    }
```

Reply:
Addressed